### PR TITLE
Added support for asymmetric algorithms in play apps

### PR DESCRIPTION
**Pull request by aleks-demin**

### CODE REVIEW

**Comment by pauldijou** [comment]
Do you know when was `play.http.session.public.key` introduced? Or any documentation about it? I totally lost track of Play updates but I would like to have a quick look to ensure I'm understanding it correctly. PR looks good.

**Comment by aleks-demin** [comment]
I Introduced `play.http.session.public.key` config in this PR.
It's not a play configuration, it's a configuration for play-jwt library.

**Comment by pauldijou** [comment]
I'm a bit concerned. `play.http.secret.key` could also be used to sign cookies or other stuff, I'm not sure I would put a private key in it... It's been a long time since I thought about all that stuff though. Would it be better to also have a `play.http.session.private.key` when using asymmetric algorithms?

Also, I would not name the config key like that, I don't think it makes sense to have a `public` namespace. I would either do `play.http.session.publicKey` (and `.privateKey` eventually). Or `play.http.session.key.public` (and `.key.private` eventually). The 1st solution seems more readable to me.

What do you think?

**Comment by aleks-demin** [comment]
Is there something wrong with signing cookies and tokens with the same key? `play.http.secret.key` remains private(secret).

I like your idea to rename config key to `play.http.session.publicKey` to remove excess `public` namespace. I'll fix it.

**Comment by pauldijou** [comment]
There is nothing wrong with it. It's just it would be blocking for people who would like to use the `secret.key` for symmetric algos for cookies, and a pair of private / public keys for JWT. I think it's worth checking for a `play.http.session.privateKey` first and then fallback to `play.http.secret.key` if not defined.

**Comment by aleks-demin** [comment]
@pauldijou Done 👍

File: play/src/main/scala/JwtSession.scala [hunk 1]
```diff
@@ -124,7 +131,7 @@ object JwtSession extends JwtJsonImplicits with JwtPlayImplicits {
     case _ => Json.obj()
   }
 
-  def defaultHeader(implicit conf: Configuration): JwtHeader = key.map(_ => JwtHeader(ALGORITHM)).getOrElse(JwtHeader())
+  def defaultHeader(implicit conf: Configuration): JwtHeader = secretKey.map(_ => JwtHeader(ALGORITHM)).getOrElse(JwtHeader())
```
Comment (pauldijou) [inline play/src/main/scala/JwtSession.scala]: Shouldn't it be `privateKey`? The goal is to put the algorithm in the header if there is a key defined.

Comment (aleks-demin) [reply play/src/main/scala/JwtSession.scala]: It depends on selected algorithm. Fixed this part.

**Comment by pauldijou** [comment]
Released in 4.3.0, thanks for your contribution.

**Pull request merged**